libs/service-portal/sessions/src/components/SessionFilter/SessionFilter.tsx

```diff
@@ -0,0 +1,139 @@
+import {
+  Accordion,
+  AccordionItem,
+  Box,
+  DatePicker,
+  Filter,
+  FilterInput,
+  useBreakpoint,
+} from '@island.is/island-ui/core'
+import { m } from '@island.is/service-portal/core'
+import * as styles from './SessionFilter.css'
+import cn from 'classnames'
+import React, { useState } from 'react'
+import { useLocale } from '@island.is/localization'
+import addYears from 'date-fns/addYears'
+
+const defaultCalState = { top: false, lower: false }
+
+interface IProps {
+  onFilterChange: (value: string) => void
+  nationalId: string
+  fromDate?: Date
+  toDate?: Date
+  handleDateChange: (date: Date, type: 'from' | 'to' | 'clear') => void
```

Comment:
Wondering if we could simplify the handling a bit in the `Sessions.tsx` if we just had a specific clear handler that would reset filter date and nationalId states value and refetch. And this would just be handling date changes.